### PR TITLE
tikv-ctl: recognize engine type by data dir

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref https://github.com/tikv/tikv/issues/14654

What's Changed:

```commit-message
recognize engine type by data dir
```
In local mode, now, tikv-ctl can only recognize engine type by tikv config (if `--config` is not specified, default tikv config will be used, so the debugger will be initialzied as v1). It means if we does not pass config path to it, it will use v1's logic to perform tikv control for v2 cluster, which can return misleading results.
As passing config path every time is not convenient, this PR makes it automatically recognize engine type by data dir passed in, which is required in local mode.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

image
data dir `tikv-11` is v1, `tikv-1` is v2.

### Release note 

```release-note
None
```

### CODE REVIEW

File: cmd/tikv-ctl/src/executor.rs [hunk 1]
```diff
@@ -56,6 +57,27 @@ pub const LOCK_FILE_ERROR: &str = "IO error: While lock file";
...
+        if let Ok(ty) = e.file_type() && ty.is_dir() {
+            if e.file_name() == "tablets" {
+                engine2 = true;
+            } else if e.file_name() == "db" {
+                engine1 = true;
+            }
+        }
+    }
+    assert_ne!(engine1, engine2);
+    if engine1 {
```
Comment (bufferflies) [inline cmd/tikv-ctl/src/executor.rs:74]: Does it need to return `raftKv` if there is no `tables` or `db`?

Comment (SpadeA-Tang) [reply cmd/tikv-ctl/src/executor.rs:74]: I think it only occurs when cluster has never been started, and in this case, it will be panic at line 73.

**Review by bufferflies — APPROVED** [review]

**Pull request merged**